backend/src/bundles/user-details/user-details.repository.ts

```diff
@@ -218,12 +146,18 @@ class UserDetailsRepository implements Repository {
         );
 
         const searchResults = await query
-            .withGraphJoined('user')
-            .where('user.role', '=', UserRole.TALENT);
+            .withGraphJoined('[user, talentBadges.[badge]]')
+            .withGraphJoined('[user, talentHardSkills]')
```

Comment:
Why do you use separate withGraphJoined, if you need both of them use just .withGraphJoined('[user, talentHardSkills, talentBadges.[badge]]');